src/shm_mod.c

```diff
@@ -558,8 +558,9 @@ sr_shmmod_modinfo_rdlock_upgrade(struct sr_mod_info_s *mod_info, uint32_t sid)
         mod = &mod_info->mods[i];
         shm_lock = &mod->shm_mod->data_lock_info[mod_info->ds];
 
-        /* upgrade only read-upgr-locked modules */
-        if (mod->state & MOD_INFO_RLOCK_UPGR) {
+        /* upgrade only required read-upgr-locked modules */
```

Comment:
Please, explain the fix (at least briefly) somewhere, here, for example. Something like
```
... and leave others read-upgr-locked to prevent their locking causing potential dead-lock
```